google/cloud/aiplatform/models.py

```diff
@@ -215,10 +229,525 @@ def upload(
+            location=self.location,
+        )
+        endpoint = self.endpoint_client.get_endpoint(name=endpoint_name)
+
+        return endpoint
+
+    @classmethod
+    def create(
+        cls,
+        display_name: str,
+        description: Optional[str] = None,
+        traffic_split: Optional[Dict] = None,
+        labels: Optional[Dict] = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> "Endpoint":
+        """Creates a new endpoint.
+
+        Args:
+            display_name (str):
+                Required. The user-defined name of the Endpoint.
+                The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            description (str):
+                Optional. The description of the Endpoint.
+            traffic_split (Dict):
+                Optional. A map from a DeployedModel's ID to the
+                percentage of this Endpoint's traffic that
+                should be forwarded to that DeployedModel.
+                If a DeployedModel's ID is not listed in this
+                map, then it receives no traffic.
+
+                The traffic percentage values must add up to
+                100, or map must be empty if the Endpoint is to
+                not accept any traffic at a moment.
+            labels (Dict):
+                Optional. The labels with user-defined metadata to
+                organize your Endpoints.
+                Label keys and values can be no longer than 64
+                characters (Unicode codepoints), can only
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
+            project (str):
+                Optional. Project to retrieve endpoint from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve endpoint from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            endpoint (Endpoint):
+                Instantiated representation of the endpoint resource.
+        """
+
+        endpoint_client = cls._instantiate_client(
+            location=location, credentials=credentials
+        )
+
+        client_class = PredictionServiceClient
```

Comment:
To handle the case of a second client we can create another class method that instantiates the prediction client.

Like so:

```python

@classmethod
def _instantiate_prediction_client(cls, location=None, credentials=None):
  return initializer.global_config.create_client(
            client_class=PredictionServiceClient,
            credentials=credentials,
            location_override=location,
            prediction_client=True,
        )
```

And in this _create method you would create the client which you would use for predict calls.:

```python
prediction_client = cls._instantiate_prediction_client(location=location, credentials=credentials)
```

Similarly, I assume, you would instantiate the prediction client in the constructor:

```python
self.prediction_client = cls._instantiate_prediction_client(location=location, credentials=credentials)
```

Reply:
if we do this, is there still a reason to have `_is_client_prediction_client`?

Reply:
I agree that we most likely do not need that flag.

Reply:
address along with predict PR.